src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -186,6 +186,11 @@ private IEnumerable<TypeInferenceInfo> InferTypeInArrowExpressionClause(ArrowExp
                     return InferTypeInBaseMethodDeclaration(arrowClause.Parent as BaseMethodDeclarationSyntax);
                 }
 
+                if (arrowClause.IsParentKind(SyntaxKind.GetAccessorDeclaration))
+                {
+                    return InferTypeInGetAccessorDeclaration(arrowClause.Parent as AccessorDeclarationSyntax);
```

Comment:
since you know this has to be an accessor, use a cast

Reply:
or pattern matching - this will work for any accessor

Reply:
Pattern matching is now used.